usaspending_api/awards/v2/views/idvs/funding_treemap.py

```diff
@@ -0,0 +1,96 @@
+from collections import OrderedDict
+from copy import deepcopy
+
+from psycopg2.sql import Identifier, Literal, SQL
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from usaspending_api.common.cache_decorator import cache_response
+from usaspending_api.common.helpers.sql_helpers import execute_sql_to_ordered_dictionary
+from usaspending_api.common.views import APIDocumentationView
+from usaspending_api.common.validator.award import get_internal_or_generated_award_id_model
+from usaspending_api.common.validator.tinyshield import TinyShield
+
+FUNDING_TREEMAP_SQL = SQL("""
+    with cte as (
+        select    award_id
+        from      parent_award
+        where     {award_id_column} = {award_id}
+        union all
+        select    cpa.award_id
+        from      parent_award ppa
+                  inner join parent_award cpa on cpa.parent_award_id = ppa.award_id
+        where     ppa.{award_id_column} = {award_id}
+    )
+    select
+        {columns}
+    from
+        cte
+        inner join awards pa on
+            pa.id = cte.award_id
+        inner join awards ca on
+            ca.parent_award_piid = pa.piid and
+            ca.fpds_parent_agency_id = pa.fpds_agency_id and
+            ca.type not like 'IDV\_%'
+        inner join financial_accounts_by_awards faba on
+            faba.award_id = ca.id
+        left outer join treasury_appropriation_account taa on
+            taa.treasury_account_identifier = faba.treasury_account_id {group_by};""")
+
+
+class IDVFundingBaseViewSet(APIDocumentationView):
+
+    """Returns File C funding totals associated with an IDV's children."""
+
+    @staticmethod
+    def _parse_and_validate_request(request: Request) -> dict:
+        return TinyShield(deepcopy([get_internal_or_generated_award_id_model()])).block(request)
+
+    @staticmethod
+    def _business_logic(request_data: dict, columns: str, group_by: str) -> list:
+        # By this point, our award_id has been validated and cleaned up by
+        # TinyShield.  We will either have an internal award id that is an
+        # integer or a generated award id that is a string.
+        award_id = request_data['award_id']
+        award_id_column = 'award_id' if type(award_id) is int else 'generated_unique_award_id'
+
+        sql = FUNDING_TREEMAP_SQL.format(
+            columns=SQL(columns),
+            award_id_column=Identifier(award_id_column),
+            award_id=Literal(award_id),
+            group_by=SQL(group_by),
+        )
+        return execute_sql_to_ordered_dictionary(sql)
+
+
+class IDVFundingRollupViewSet(IDVFundingBaseViewSet):
+
```

Comment:
Should we resurrect providing the `endpoint_doc` in docstrings to point to the API Blueprint?

example: https://github.com/fedspendingtransparency/usaspending-api/blob/dev/usaspending_api/search/v2/views/search.py#L30

Reply:
This is my fault.  I did not include one in the endpoint this is based upon because we were in the middle of transitioning API docs over at that time.

Would this be a good time to update the code that translates `endpoint_doc` to also look for documents in the new API Contract directory so we can start migrating rather than maintaining two document bases?